src/fidesops/tasks/storage.py

```diff
@@ -21,38 +31,79 @@
 logger = logging.getLogger(__name__)
 
 
-def write_to_in_memory_buffer(resp_format: str, data: Dict[str, Any]) -> BytesIO:
-    """Write JSON/CSV data to in-memory file-like object to be passed to S3
+LOCAL_FIDES_UPLOAD_DIRECTORY = "fides_uploads"
+
+
+def encrypt_access_request_results(data: Union[str, bytes], request_id: str) -> str:
+    """Encrypt data with encryption key and nonce (if provided) using AES GCM, otherwise return unencrypted data"""
+    cache = get_cache()
+    encryption_cache_key = get_encryption_cache_key(
+        privacy_request_id=request_id,
+        encryption_attr="key",
+    )
+    if isinstance(data, bytes):
+        data = data.decode(config.security.ENCODING)
+
+    encryption_key = cache.get(encryption_cache_key)
+    if not encryption_key:
+        return data
+
+    encryption_key = encryption_key.encode(encoding=config.security.ENCODING)
+    nonce = cryptographic_util.generate_secure_random_string(
+        math.floor(config.security.AES_GCM_NONCE_LENGTH / 2)
+    )
```

Comment:
This feels a little weird - our `generate_secure_random_string` method currently generates a string that is twice as long as the integer we use as the input, but the `verify_nonce` method used by our internal aes gcm `encrypt` requires that the nonce be `AES_GCM_NONCE_LENGTH` in length, so I halve that value here.

Reply:
This is related to https://github.com/ethyca/fidesops/issues/83, for a similar reason our AES Encryption/Decryption endpoints are broken.

Reply:
I've run into this exact problem with the `secrets.token_***` methods in other projects, they're slightly confusing. For `token_hex` in particular (which is what our crypto util is using) it generates a N-byte value and then converts that to hex- with two hex characters per byte. So what you really need to do is probably just use `secrets.token_bytes`, for example: https://docs.python.org/3/library/secrets.html

As an alternative, note that `os.urandom` might be a better fit for this as it's what most projects seem to use for Python crypto, but that might just be due to timing - I think the `secrets` module is a newer addition that is designed to replace the need for that.

Reply:
Also FYI - the nonce length *needs* to be that exact length for the AES algorithm to work (if you go ahead and try it with other lengths and remove that `verify_nonce` check it should throw an error!). For that reason, I don't think it really should be a config option either.

Reply:
summary of separate discussion, I'm now using `secrets.token_bytes` and using b64 to encode the nonce and the encrypted data together to return to the user, thanks @NevilleS 

```python
iv = secrets.token_bytes(12)
ciphertext = <insert binary string here>
output = b64encode(iv + ciphertext)
```

```python
decoded = b64decode(output)
iv = decoded[0:12]
cipherext = decoded[12:]
```